go/worker/keymanager/worker.go

```diff
@@ -178,32 +184,23 @@ func (w *Worker) callLocal(ctx context.Context, data []byte) ([]byte, error) {
 }
 
 func (w *Worker) updateStatus(status *api.Status, startedEvent *host.StartedEvent) error {
-	var initOk bool
 	defer func() {
-		if !initOk {
-			// TODO: once #2130 is done and keymanager reports as Ready only
-			// after initialization succeeds, change this to always pre-register
-			// on the initial updateStatus call (and not only after the first
-			// initialization fails as it is done currently).
-			// This is likely a new key manager that needs to replicate.
-			// Send a node registration anyway, so that other nodes know
-			// to update their access control.
-			w.roleProvider.SetAvailable(func(n *node.Node) error {
-				rt := n.AddOrUpdateRuntime(w.runtime.ID())
-				rt.Version = startedEvent.Version
-				rt.ExtraInfo = nil
-				rt.Capabilities.TEE = startedEvent.CapabilityTEE
-				return nil
-			})
-
-			// If initialization failed setup a retry ticker.
-			if w.initTicker == nil {
-				w.initTicker = backoff.NewTicker(backoff.NewExponentialBackOff())
-				w.initTickerCh = w.initTicker.C
-			}
+		// If initialization failed setup a retry ticker.
+		if w.initTicker == nil {
+			w.initTicker = backoff.NewTicker(backoff.NewExponentialBackOff())
+			w.initTickerCh = w.initTicker.C
 		}
 	}()
 
+	// Pre-register to allow replication.
+	w.roleProvider.SetAvailable(func(n *node.Node) error {
```

Comment:
I think it might make more sense to move this out of `updateStatus` function, probably to where we watch for the started event and do it once we receive the started event. (that way this will also be done only once, and not on every status update - which is i think what we want)